### DISCUSSION TITLE
Getting Error EBUSY: resource busy or locked - pnpm install

### DISCUSSION THREAD

**Discussion by sharonanderson-dev** [community/community#50143] [Codespaces] [answered]
### Select Topic Area

Question

### Body

I tried to install dependencies from `Package.json` file with `pnpm install` but it's not working I tried in several ways but I can't find a way to solve it.
Here is my debbuged error message.
```javascript
Error: EBUSY: resource busy or locked, write
at Object.writeSync (fs.js:568:3)
at Object.writeFileSync (fs.js:1199:26)
at new Gpio (/home/pi/Desktop/pitesting/node_modules/onoff/onoff.js:96:10)
at Object.<anonymous> (/home/pi/Desktop/pitesting/blink.js:3:7)
at Module._compile (internal/modules/cjs/loader.js:701:30)
at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
at Module.load (internal/modules/cjs/loader.js:600:32)
at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
at Function.Module._load (internal/modules/cjs/loader.js:531:3)
at Function.Module.runMain (internal/modules/cjs/loader.js:754:12)
```

**Comment by santosharron** [comment, marked as answer]
Okay this error message usally shows up on windows when something goes wrong while installing the package that's has already been installed. So in order to fix this issue **I'll recommend you to clean the temporary files on your pc**.
Head over to search bar and search for this term `%temp%` and delete all the files. Don't worry it's just the temporary files that's been stored on your computer for managing the data. If you delete it nothing will happen so go ahead and do that.

There is another method of solving this but it has been deprecated but I'll tell you how that works as well. Basically you need to clear the temporary data anyways so in order to do that using npm you can run the following command `npm cache clean`.

Hope it helps :)

**Reply by sharonanderson-dev** [reply]
Thanks it worked!

**Reply by animeshdas2000** [reply]
I tried following the above steps, but it didn't work out. Apparently, I was on a different version than the node engine supported by the repository. Changing the node version did help. Leaving this answer here for someone if they get blocked.

**Reply by PatrykKuniczak** [reply]
Thanks :)

**Reply by cope** [reply]
@animeshdas2000
> Changing the node version did help.

Right, but to what version and how did you figure out which version to switch to? :(

**Reply by sijujacobs** [reply]
Here's a workaround for similar issue with StencilJS
https://github.com/ionic-team/stencil/issues/5486

**Comment by diraneCode** [comment]
Quand tu rencontres cette erreur commence par redemarrer ton pc, si cela ne fonctionne pas toujours essaie de réinstaller node js et pnpm à nouveau

**Comment by abhisheek227** [comment]
```
EBUSY: resource busy or locked, rm 'C:\Users\ABHISH~1\AppData\Local\Temp\lighthouse.83188449'
    path: "C:\\Users\\ABHISH~1\\AppData\\Local\\Temp\\lighthouse.83188449",
 syscall: "rm",
   errno: -16,
    code: "EBUSY"
```

hey i am getting this error can any one tell me how i can solve this one?
